src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -390,14 +389,62 @@
 
             if (validIssuerTemplate.Contains(TenantIdTemplate))
             {
-                return validIssuerTemplate.Replace(TenantIdTemplate, tenantId) == actualIssuer;
+                return IssuersWithTemplatesAreEqual(validIssuerTemplate, TenantIdTemplate, actualIssuer, tenantId);
             }
             else
             {
                 return validIssuerTemplate == actualIssuer;
             }
         }
```

Comment:
>firstHalfIssuer2EndIndex [](http://example.com/codeflow?start=16&length=24)

There is no need for this variable since it is only used when compared with the firstHalfIssuer1EndIndex. You may directly include the IndexOf as part of the if statement below and just keep a single variable. 

After this check both firstHalfIssuer1EndIndex and firstHalfIssuer2EndIndex will have the same value.

Reply:
Thanks, done